Add default gem seedlist docs (#2431)
o3de/o3de.org: content/docs/user-guide/programming/gems/creating.md

Each Gem has an `Assets` directory that can contain models, textures, scripts, animations, and more. Asset files are accessed the same way as they are in a game project. O3DE uses this root directory to find the asset file path. For example, when O3DE looks for the `textures/rain/rainfall_ddn.tif` file, it looks in the `<GemName>/Assets/textures/rain/` directory.

Assets used by your Gem in a launcher runtime should have their source assets listed in a [Gem dependencies `.xml` file](#gem-dependencies) or their compiled asset path should be listed in [`<GemName>/Assets/seedList.seed`](#default-gemnameassetsseedlistseed). 

### Default `<GemName>/Assets/seedList.seed`

> This file contains a list of compiled asset paths and platforms that will automatically appear in the *Asset Bundler* GUI making it easy for users to find and include in their project asset bundle.

Reviewer: ```suggestion
This file contains a list of compiled asset paths and platforms that will automatically appear in the Asset Bundler GUI making it easy for users to find and include in their project asset bundle.
```

> They will also be included in a bundle when the [*AssetBundlerBatch* `--addDefaultSeedListFiles`](/docs/user-guid/packaging/asset-bundler/command-line-reference/#options) CLI option is used.

Reviewer: ```suggestion
They will also be included in a bundle when the [*AssetBundlerBatch* `--addDefaultSeedListFiles`](/docs/user-guide/packaging/asset-bundler/command-line-reference/#options-2) CLI option is used.
```

The `seedList.seed` approach is useful when you want to provide a ready-to-use list of default compiled assets and the platforms they're used on.

> Refer to the [Asset Bundler documentation](/docs/user-guide/packaging/asset-bundler/) for more information on the *Asset Bundler*, `.seed` files and bundling assets,

Reviewer: ```suggestion
Refer to the [Asset Bundler documentation](/docs/user-guide/packaging/asset-bundler/) for more information on the *Asset Bundler*, `.seed` files, and bundling assets.
```

{{<important>}}

> The filename `seedList.seed` is case-sensitive and must be in your Gem `Assets` folder or the seed list will not automatically appear in the *Asset Bundler* GUI.

Reviewer: ```suggestion
The filename `seedList.seed` is case-sensitive and must be in your Gem `Assets` folder or the seed list will not automatically appear in the Asset Bundler GUI.
```

{{</important>}}

### Gem dependencies

> The Gem dependencies `.xml` file contains a list of source assets which will be used by the *Asset Processor* to create a seed list containing all the compiled asset paths and platforms.

Reviewer: ```suggestion
The Gem dependencies `.xml` file contains a list of source assets which will be used by the Asset Processor to create a seed list containing all the compiled asset paths and platforms.
```
